Submodule warnings/misc

colin-daniels opened the pull request:
Updated readme and merged warning for when git submodule isn't checked out. The error it threw before was not particularly helpful. Also added `shallow = true` to the lammps submodule, which...may or may not actually help when updating.

joelgallant committed:
Adds warning when git submodule wasn't checked out

colin-daniels committed:
Merge remote-tracking branch 'fork/master'

# Conflicts:
#	build/build.rs

colin-daniels committed:
add shallow to lammps submodule

The lammps git is huge, so using shallow is good (note: supported
since git 2.10).

colin-daniels committed:
update README with submodule info

colin-daniels committed:
fix submodule error text

colin-daniels committed:
update error text some more

colin-daniels committed:
Revert "update error text some more"

This reverts commit c98e0496. I didn't even try to build...

ExpHP committed:
Document why first line succeeds

ExpHP committed:
add clarification

ExpHP commented:
Yep, this mistake in the instructions was missed for the same reason as #10; I had a local install of lammps while I was testing them out!

Looks good to me.

ExpHP commented:
Actually, I'm not sure about the `shallow = true`.

As far as I can tell it doesn't seem to do anything.  From what I'm reading online, it should cause `lammps/` to be a shallow clone unless `--no-recommend-shallow` is specified.  This sounds reasonable (you only need a deep clone if you want to check out a different version of lammps)?  However, I tried to test it using `git submodule init --update` or `git clone --recurse-submodules` on this PR, and no matter what I do `lammps/` always ends up being a 334MB repo with a complete history.

I even tried `git clone --recurse-submodules --shallow-submodules`, which still had the same outcome.  (though if that *did* work, it sounds much safer to have the docs recommend this command, where somebody can see it and know why the submodule is shallow)

Maybe it doesn't work when the revision to be checked out is a tag?

colin-daniels commented:
Yeah I'm not sure myself, I usually use `--depth 1` when cloning big repos like this which does do what you expect, but for submodules here I'm not sure what's going on (initially I thought it helped, but it was just git caching something I think?). If there's no actual difference as far as we can tell, might as well just remove `shallow = true` so there's no confusion of what is or isn't happening.

ExpHP committed:
undo the shallow submodule

It doesn't seem to do anything:

https://github.com/ExpHP/lammps-sys/pull/12#issuecomment-493792480
